### PR TITLE
Move Datalogger to proto3, slight format changes

### PR DESCRIPTION
**Pull request by ducky64**
Deprecate temperature, voltage, and loop timer payload types in favor of unified sensorReading,

### CODE REVIEW

File: datalogger/datalogger.proto [hunk 1]
```diff
@@ -37,12 +37,14 @@ message DataloggerPayload {
     CanMessage receivedCanMessage = 3;
     CanMessage transmittedCanMessage = 4;
     CanError canError = 5;
-    CanErrorCounter canErrorCount = 6;
-    StatisticalAggregate voltageReading = 7;
-    StatisticalAggregate temperatureReading = 8;
-    StatisticalAggregate loopTimer = 9;
-    IntHistogram loopTimerDistribution = 11;
+    CanErrorCounter canErrorCount = 6 [deprecated=true];
```
Comment (avi-jois) [inline datalogger/datalogger.proto:40]: Are you marking these as deprecated so we still have compatibility with older data?

**Review by avi-jois — APPROVED** [review]
LGTM, but I think we should remove the generated files from the repo and have a scons hook to generate them on compile time

File: datalogger/datalogger.proto (hunk 1 above)
Comment (ducky64) [reply datalogger/datalogger.proto:40]: Yeah, they will still be accessible (so datalogger parser can run over the fsgp data) but discouraged from use in new data. Not that nanopb generates deprecation warnings, sadly.

Comment (avi-jois) [reply datalogger/datalogger.proto:40]: OK, just checking

**Comment by ducky64** [comment]
The idea behind generated files being committed is that most users who need to build firmware (instead of developing protos) shouldn't need to have protoc installed, and can instead rely on the committed generated files.

There's already a scons option in the main firmware repo that will (re)build the common-proto submodule. Off by default, for the reasons above.

**Pull request merged**